test/e2e/testing-manifests/storage-csi/gce-pd/node_ds.yaml

```diff
@@ -0,0 +1,72 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: csi-gce-node
+spec:
+  selector:
```

Comment:
Why does this need a Pod Selector?

Reply:
I think its necessary: https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#pod-selector

Reply:
Ok, I wonder why that is.

Reply:
I think it somehow "ties" the higher level workload to the pods that it controls. It seems that previous behavior was that if not specified it would default, but since the "tying" wasn't explicit people were confused/breaking things when modifying their API objects afterwards and the pods would get orphaned